types/node/net.d.ts

```diff
@@ -71,7 +71,7 @@ declare module "net" {
         setTimeout(timeout: number, callback?: () => void): this;
         setNoDelay(noDelay?: boolean): this;
         setKeepAlive(enable?: boolean, initialDelay?: number): this;
-        address(): AddressInfo | string;
+        address(): AddressInfo;
```

Comment:
```suggestion
        address(): AddressInfo | {};
```
In case of windows pipes or unix domain sockets `{}` is returned (server returns the path as string).

Reply:
This makes sense, also when some data is not available `{}` is returned. From source code: https://github.com/nodejs/node/blob/92610f330d6e149f782d9d1cdf106cd629374de4/lib/net.js#L714

Will update PR